vllm/lora/layers.py

```diff
@@ -668,25 +682,27 @@ def create_lora_weights(
             max_loras: int,
             lora_config: LoRAConfig,
             model_config: Optional[PretrainedConfig] = None) -> None:
+        device = _get_lora_device(self.base_layer)
+
         self.lora_a_stacked = torch.zeros(
             (
                 max_loras,
                 1,
                 lora_config.max_lora_rank,
-                self.base_layer.weight.shape[1],
+                self.base_layer.input_size,
             ),
             dtype=lora_config.lora_dtype,
-            device=self.base_layer.weight.device,
+            device=device,
         )
         self.lora_b_stacked = torch.zeros(
             (
                 max_loras,
                 1,
-                self.base_layer.weight.shape[0],
+                self.base_layer.output_size,
                 lora_config.max_lora_rank,
             ),
             dtype=lora_config.lora_dtype,
-            device=self.base_layer.weight.device,
+            device=device,
         )
         self.indices: Optional[torch.Tensor] = None
         self.indices_len: Optional[List[int]] = None
```

Comment:
@fmmoret  Thank you for your contribution. I have tested this pull request, and the result looked good. However, one issue that needs to be considered is tensor parallelism.

Reply:
Just change `self.base_layer.input_size` to `self.base_layer.input_size_per_partition` for `RowParallelLinear` module, and change `self.base_layer.output_size` to `self.base_layer.output_size_per_partition` for `ColumnParallelLinearWithLoRA`. We could run on tensor parallelism setting using this PR.

Reply:
Also, there might raise NoneType error when tp_size > 1 and either lora_b[0] or lora_b[1] is None. It's still a bug in main branch

```python
if self.tp_size > 1:
            tensor_model_parallel_rank = get_tensor_model_parallel_rank()
            shard_size = self.output_dim
            start_idx = tensor_model_parallel_rank * shard_size
            end_idx = (tensor_model_parallel_rank + 1) * shard_size
            # lora_b = lora_b[0][:,
            #                    start_idx:end_idx], lora_b[1][:,
            #                                                  start_idx:end_idx]
            if lora_b[0] is not None:
                lora_b[0] = lora_b[0][:, start_idx:end_idx]
            if lora_b[1] is not None:
                lora_b[1] = lora_b[1][:, start_idx:end_idx]
```